app/javascript/views/publicChannelPage/CryptoPaymentWidget.jsx

```diff
@@ -1,20 +1,360 @@
+        options: [
+          { label: intl.formatMessage({ id: 'walletServices.addCryptoWidget.solana' }), value: "SOL", icon: solIcon },
+        ]
+      })
+    }
+
+    const currentChain = ethAddress ? 'ETH' : 'SOL';
+
+    this.state = {
+      placeholder,
+      isLoading: true,
+      currentAmount: 5,
+      addresses,
+      dropdownOptions,
+      // the channel must have at least one crypto address for this page to be navigable,
+      // and right now the options are only sol and eth
+      currentChain,
+      defaultAmounts: [1,5,10],
+      isModalOpen: false,
+      ratios: {},
+      customAmount: null,
+      toggle: 'crypto',
+      selectValue: dropdownOptions.flatMap(opt => opt.options).filter(opt => opt.value === currentChain)[0],
+      isSuccessView: false,
+    }
+  }
+
+  componentDidMount() {
+    this.loadData();
+  }
+
+  loadData = () => {
+    this.setState({ isLoading: true });
+    axios.get(routes.publishers.publicChannelPage.getRatios).then((response) => {
+      const newState = { ...this.state }
+      newState.isLoading = false;
+      newState.ratios = response.data;
+      this.setState({ ...newState });
+    });
+  };
+
+  calculateCryptoPrice() {
+    return this.state.currentAmount / this.state.ratios[this.state.currentChain.toLowerCase()]['usd'];
+  }
+
+  roundCryptoPrice() {
+    return Math.round(this.calculateCryptoPrice() * 100000) / 100000;
+  }
+
+  closeModal = () => {
+    this.setState({ isModalOpen: false });
+  }
+
+  launchQRModal() {
+    this.setState({ isModalOpen: true });
+  }
+
+  sendPayment = async () => {
+    if (this.state.currentChain === "ETH") {
+      await this.sendEthPayment();
+    } else if (this.state.currentChain === "SOL") {
+      this.sendSolPayment();
+    }
+  }
+
+  sendEthPayment = async () => {
+    if (window.ethereum) {
+      const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' })
+      const address = accounts[0]
+      if (!address) {
+        // set to be designed error state here
+        return;
+      }
+
+      // While most guides to converting eth to wei multiply the value by 10e18, In javascript e counts 
+      // as the 10 and *10e18 results in a value that is an order of mangitude too high.
+      const value = Web3Utils.toHex(Web3Utils.toBigInt(Math.round(this.calculateCryptoPrice()*10e17)));
+
+      const params = [{
+        from: address,
+        to: this.state.addresses.ETH,
+        value: value
+      }];
+
+      const transaction = window.ethereum
+        .request({
+          method: 'eth_sendTransaction',
+          params,
+        })
+        .then((result) => {
+          // window.ethereum.disable()
+          const newState = {...this.state};
+          newState.isSuccessView = true;
+          this.setState({...newState });
+        })
+        .catch((error) => {
+          console.log(error)
+          // window.ethereum.disable()
```

Comment:
Do we want to leave this in?

Reply:
We can get rid of it.  Apparently window.ethereum.disable() is spec'd out, but hasn't yet been implmented.  I'm hoping to find another way for the page to close the connection with the wallet in the future.